test/timeout.js

```diff
@@ -0,0 +1,210 @@
+
+  const started = Date.now()
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 50), 'timers took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50)
+})
+
+test('interrupt setTimeout with CPU overhead', async function (t) {
+  t.plan(1)
+
+  const started = Date.now()
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 75), 'timers took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50)
+
+  while (Date.now() - started < 75) {} // eslint-disable-line no-empty
+
+  /* setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 125), 'native took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50) */
+})
+
+test('interrupt setTimeout with Atomics.wait', async function (t) {
+  t.plan(1)
+
+  const started = Date.now()
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 75), 'timers took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50)
+
+  sleep(75)
+
+  /* setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 125), 'native took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50) */
+})
+
+test('multiple setTimeout', async function (t) {
+  t.plan(4)
+
+  const started = Date.now()
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 20), '1st timer took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 20)
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 50), '2nd timer took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 50)
+
+  timers.setTimeout(function () {
+    t.ok(isAround(Date.now() - started, 20), '3rd timer took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 20)
+
+  timers.setTimeout(() => {
+    t.ok(isAround(Date.now() - started, 0), '4th timer took ' + Math.abs(Date.now() - started) + 'ms')
+  }, 1)
+})
+
+test('clearTimeout', async function (t) {
+  t.plan(1)
+
+  const id = timers.setTimeout(() => t.fail('timeout should not be called'), 20)
+  setTimeout(() => timers.clearTimeout(id), 15)
+  setTimeout(() => t.pass(), 50)
+})
+
+test('clearTimeout twice', async function (t) {
+  t.plan(1)
+
+  const id = timers.setTimeout(() => t.fail('timeout should not be called'), 20)
+  setTimeout(() => {
+    timers.clearTimeout(id)
+    timers.clearTimeout(id)
+  }, 15)
+  setTimeout(() => t.pass(), 50)
+})
+
+test('lots of setTimeout + clearTimeout', async function (t) {
+  t.plan(1)
+
+  const timeouts = new Array(2000000)
+  let pass = 0
+
+  for (let i = 0; i < timeouts.length; i++) {
+    timeouts[i] = timers.setTimeout(ontimeout, 10)
+    if (i % 2 === 0) timers.clearTimeout(timeouts[i])
+  }
+
+  function ontimeout () {
+    if (++pass === timeouts.length / 2) {
+      t.pass()
+    }
+  }
+})
+
+test('error inside of setTimeout', async function (t) {
+  t.plan(2)
+
+  const error = new Error('random')
+
+  timers.setTimeout(function () {
+    t.pass()
+    throw error
+  }, 10)
+
+  process.once('uncaughtException', function (err) {
+    t.is(err, error)
+  })
+})
```

Comment:
can you schedule a later timeout here also and see that it triggers, even after the uncaught has triggered?